src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+    public final static int kMaxProbePackets = 15;
+    public final static int kExpectedNumberOfProbes = 3;
+
+    private static final double kTimestampToMs = 1000.0 /
+            (1 << kInterArrivalShift) ;
+
+    public <K,V> List<K> Keys(TreeMap<K,V> _map)
+    {
+        ArrayList<K> keys = new ArrayList<K>();
+        for(Map.Entry<K,V> entry : _map.entrySet())
+        {
+            keys.add(entry.getKey());
+        }
+        return keys;
+    }
+
+    long ConvertMsTo24Bits(long timeMs)
+    {
+        long time24Bits = (long)(((timeMs << kAbsSendTimeFraction) + 500) /
+                1000) & 0x00FFFFFF;
+        return time24Bits;
+    }
+
+    /**
+      * Confirm this on line 133 of remote_bitrate_estimator_abs_send_time.h
+     */
+    public TreeMap<Long,Long> ssrcs_ = new TreeMap<Long, Long>();
+    public static ArrayList<Probe> probes_ = new ArrayList<>();
+    public static long totalProbesReceived;
+    public static  long firstPacketTimeMs;
+    public static long lastUpdateMs;
+    public static boolean umaRecorded;
+
+    /**
+     * @Todo confirm these
+     * clock and observer_ are defined as const
+     * e.g const Clock* const clock_;
+     * find out if there is need to make it final
+     */
+    public static long clock;
+    public static RemoteBitrateObserver observer_;
+
+
+    private AimdRateControl remoteRate  = new AimdRateControl();
+    private InterArrival interArrival;
+    private OveruseEstimator estimator;
+    private OveruseDetector detector;
+    private OverUseDetectorOptions options;
+    private RateStatistics incomingBitrate ;
+    boolean incomingBitrateInitialized;
+    ArrayList<Integer> recentPropagationDeltaMs;
+    ArrayList<Long> recentUpdateTimeMs;
+
+
+    boolean IsWithinClusterBounds(long sendDeltaMs, Cluster clusterAggregate)
+    {
+        if(clusterAggregate.count == 0)
+        {
+            return true;
+        }
+        double clusterMean = clusterAggregate.sendMeanMs /
+                (double)clusterAggregate.count;
+        /**
+         * @TODO what is {@link fabs} ? No reference in webrtc absendtime.
+         */
+        return  Math.abs((double)sendDeltaMs - clusterMean) < 2.5f;
+    }
+
+    List<Cluster> addCluster(List<Cluster> clusters, Cluster cluster)
+    {
+        cluster.sendMeanMs /= (double)cluster.count;
+        cluster.recvMeanMs /= (double)cluster.count;
+        cluster.meanSize /= cluster.count;
+        clusters.add(cluster);
+        return clusters;
+    }
+
+    /**
+     * @TODO Revisit line 81 of https://chromium.googlesource.com
+     * /external/webrtc/+/master/webrtc/modules/remote_bitrate_estimator
+     * /remote_bitrate_estimator_abs_send_time.cc
+     */
+
+    void RemoteBitrateEstimatorAbsSendTime(RemoteBitrateObserver observer,
+                                           long time)
+    {
+        this.observer_ = observer;
+        this.clock = time;
+        this.interArrival = new InterArrival(90 * kTimestampGroupLengthMs,
```

Comment:
`this.clock` will be removed. `observer` is passed as a parameter when instantiating RBE_abs_send_time (as in webrtc).
the `InterArrival` parameters are default values. The same parameters were used in the RemoteBitrateEstimatorSingleStream. Also definition on line 34 of [Webrtc](https://chromium.googlesource.com/external/webrtc/+/master/webrtc/modules/remote_bitrate_estimator/inter_arrival.h )